src/DropboxOauthButton.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import CloudUploadIcon from '@material-ui/icons/CloudUpload';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = theme => ({
+    button: {
+        margin: theme.spacing.unit,
+    },
+    leftIcon: {
+        marginRight: theme.spacing.unit,
+    },
+    rightIcon: {
+        marginLeft: theme.spacing.unit,
+    },
+    iconSmall: {
+        fontSize: 20,
+    },
+});
+
+const DropboxOauthButton = ({ getToken, isAuthenticated, classes }) => {
+    const [dialogOpen, setDialogOpen] = useState(false);
+    const [inputText, setInputText] = useState(undefined);
```

Comment:
Use `null` instead of undefined. Otherwise also, if nothing is passed as param, default value should be undefined.